tensorboard/webapp/plugins/npmi/views/main/main_container_test.ts

```diff
@@ -18,37 +18,118 @@ limitations under the License.
     await TestBed.configureTestingModule({
       declarations: [MainComponent, MainContainer],
-      imports: [RunsModule],
-      providers: [provideMockStore({})],
+      imports: [],
+      providers: [
+        provideMockStore({
+          initialState: {
+            ...createState(createCoreState()),
+            ...appStateFromNpmiState(createNpmiState()),
+          },
+        }),
+      ],
     }).compileComponents();
     store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
-    store.overrideSelector(getRunSelection, new Map([['run_1', true]]));
+
+    dispatchedActions = [];
+    spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      dispatchedActions.push(action);
+    });
   });
 
-  it('renders npmi main component', () => {
+  it('renders npmi main component without runs', () => {
+    store.overrideSelector(getRunSelection, new Map());
     const fixture = TestBed.createComponent(MainContainer);
     fixture.detectChanges();
 
     const runsElement = fixture.debugElement.query(
       By.css('tb-legacy-runs-selector')
     );
     expect(runsElement).toBeTruthy();
+
+    const analysisElement = fixture.debugElement.query(
+      By.css('.analysis-container')
+    );
+    expect(analysisElement).toBeNull();
+  });
+
+  it('renders npmi main component with run', () => {
+    store.overrideSelector(getRunSelection, new Map([['run_1', true]]));
+    const fixture = TestBed.createComponent(MainContainer);
+    fixture.detectChanges();
+
+    const runsElement = fixture.debugElement.query(
+      By.css('tb-legacy-runs-selector')
+    );
+    expect(runsElement).toBeTruthy();
+
+    const analysisElement = fixture.debugElement.query(
+      By.css('.analysis-container')
+    );
+    expect(analysisElement).toBeTruthy();
+  });
+
+  it('dispatches sidebar toggle when disabled and toggle button clicked', () => {
+    store.overrideSelector(getSidebarExpanded, false);
+    const fixture = TestBed.createComponent(MainContainer);
+    fixture.detectChanges();
+
+    const sidebarElement = fixture.debugElement.query(
+      By.css('.sidebar-container')
+    );
+    expect(sidebarElement).toBeNull();
+    const sideToggle = fixture.debugElement.query(By.css('.side-toggle'));
+    expect(sideToggle).toBeTruthy();
+    const expansionButton = sideToggle.query(By.css('button'));
+    expansionButton.nativeElement.click();
+
+    expect(dispatchedActions).toEqual([
+      npmiActions.npmiToggleSidebarExpanded(),
+    ]);
+  });
+
+  it('renders sidebar and grabber when enabled', () => {
+    const fixture = TestBed.createComponent(MainContainer);
+    fixture.detectChanges();
+
+    const sidebarElement = fixture.debugElement.query(
+      By.css('.sidebar-container')
+    );
+    expect(sidebarElement).toBeTruthy();
+    const grabberElement = fixture.debugElement.query(By.css('.grabber'));
+    expect(grabberElement).toBeTruthy();
+  });
+
+  it('dispatches change sidebarWidth when interacted with grabber', () => {
```

Comment:
You also have logic ensuring the sidebar width changes are not dispatched under certain circumstances. Do you want to add tests for those?

Reply:
Added test.